uvdat/core/rest/networks.py

```diff
@@ -0,0 +1,40 @@
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import serializers
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
+
+from uvdat.core.models import Network
+from uvdat.core.rest.access_control import GuardianFilter, GuardianPermission
+from uvdat.core.rest.serializers import NetworkSerializer
+
+
+class GCCQueryParamSerializer(serializers.Serializer):
+    exclude_nodes = serializers.RegexField(r'^\d+(,\s?\d+)*$')
+
+
+class GCCResultSerializer(serializers.Serializer):
+    gcc = serializers.ListField(child=serializers.IntegerField())
+
+
+class NetworkViewSet(ModelViewSet):
+    queryset = Network.objects.all()
+    serializer_class = NetworkSerializer
+    permission_classes = [GuardianPermission]
+    filter_backends = [GuardianFilter]
+    lookup_field = 'id'
+
+    @swagger_auto_schema(query_serializer=GCCQueryParamSerializer)
+    @action(detail=True, methods=['get'])
+    def gcc(self, request, **kwargs):
+        network = self.get_object()
+
+        # Validate and de-serialize query params
+        serializer = GCCQueryParamSerializer(data=request.query_params)
+        serializer.is_valid(raise_exception=True)
+        exclude_nodes = [int(n) for n in serializer.validated_data['exclude_nodes'].split(',')]
+
+        gcc = network.get_gcc(excluded_nodes=exclude_nodes)
+        result = GCCResultSerializer(data=dict(gcc=gcc))
+        if result.is_valid():
+            return Response(result.data.get('gcc'), status=200)
```

Comment:
While we're unlikely to hit this case, since this is a trivial serializer, but if the result _weren't_ valid, this endpoint would error, as it's returning nothing.

I think it's better to explicitly invoke this case, like so:

```suggestion
        result.is_valid(raise_exception=True)
        return Response(result.validated_data['gcc'], status=200)
```

This will raise an exception if the data input is malformed.

Reply:
I'll also point out that generally speaking the serializer defines the response shape, so you can imagine having a response structure of something like:

```
{
  "count": 10,
  "gcc": [
    1,
    2,
    ...
  ]
} 
```

that you would just pass to the response like so:

```
return Response(result.validated_data)
```

This is probably beyond the scope of this PR, but I just thought I'd mention it.

Reply:
Ok, I applied your suggested change. And I see what you mean about the return value, but for this PR I tried not to change the response structure so it would not affect the client. We can return to that later if necessary.